src/Generator.Bind/Writers/BindingWriter.cs

```diff
@@ -160,12 +160,12 @@ private Task WriteExtensionWrapperFiles([NotNull] string extensionName, [NotNull
             }
 
             var allCategories = _profile.GetCategories();
-            var categoryTasks = allCategories.Select(category => WriteCategoryWrapperFile(extensionOutputDirectory, extensionName, category));
+            var categoryTasks = allCategories.Select(category => Task.Run(() => WriteCategoryWrapperFile(extensionOutputDirectory, extensionName, category)));
 
             return Task.WhenAll(categoryTasks);
         }
 
-        private async Task WriteCategoryWrapperFile
+        private void WriteCategoryWrapperFile
```

Comment:
I'd like to see before and after numbers on this change. The intent is to allow asynchronous writing of multiple files at once, and this change makes this method strictly synchronous. 

With the proposed changes, I would expect performance to drop for multi-target runs.

Reply:
You can't parallelize writing to a single file; it's a fundamentally synchronous operation because the order of the text you're writing matters. What we can do is run multiple of these synchronous file IO operations as parallel tasks, via Task.Run.

By eliminating a bajillion task invocations via the await keyword, we're able to get faster performance within each of these tasks, and better tuning of our parallelism.

Some benchmarks:

Before:
```
Bind.exe
OpenGL binding generator 1.0.0.0 for OpenTK.
For comments, bugs and suggestions visit http://github.com/opentk/opentk

Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics

Bindings generated in 12.3194237 seconds.

Process finished with exit code 0.
```

After:

```
>Bind.exe
OpenGL binding generator 1.0.0.0 for OpenTK.
For comments, bugs and suggestions visit http://github.com/opentk/opentk

Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics
Writing bindings to src\OpenTK\Graphics

Bindings generated in 8.288314 seconds.

Process finished with exit code 0.
```

Reply:
Why not Task WriteText(string[])?

Reply:
I actually tried using pre-allocated buffers and then writing the files at once. This surprisingly was slower due to the fact that each line had to be written twice now. It turns out that .Net is doing a really good job of buffering the file IO to make it as fast as possible.

Reply:
Are these tests with or without the license text change? The idea isn't that a single file should be parallelized - it's obviously not, what with `await` all over the place - but rather, writing to different files in potentially different targets in parallel.

Reply:
These benchmarks are with all the changes. 

The file IO operation is serial, but we execute it for each file as it's own task in parallel.
I'm using a multi-core with all targets enabled and the performance has improved.

The reason why it's faster is very simple:
Writing each line via async/await is very inefficient. It would involve creating task objects, indirection and possibly context switches. It's a serial operation operation that should be done by one task (per file), not one task per line.

I don't have time to separate out the license changes and benchmark them for an offline tool, but it's pretty obvious that hitting the disk each time we write a file is not great for performance.

Reply:
I ran the tests myself, and it is indeed significantly faster. Sorry for being so difficult on this one :P Whenever there's a change that gets made for perf reasons, I'm always very interested to see why and how much of a difference there is.